File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.java
import java.util.*;
class Solution {
    public int[][] solution(int[][] arr) {
        int[][] answer = {};
        
        int index = 0;
        
        for(int i = 0; i < arr.length; i++){
            index = arr[i].length < arr.length ? arr.length : arr[i].length; 
        }
        
        answer = new int[index][index];
        
        for(int j = 0; j < arr.length; j++){
            for(int k = 0; k < arr[j].length; k++){
                answer[j][k] = arr[j][k];
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.py
import copy
def solution(arr):
    answer = copy.deepcopy(arr)
    
    if len(arr) > len(arr[0]):
        for i in range(len(answer)):
            answer[i] += [0 for _ in range(len(arr) - len(arr[0]))]

    elif len(arr) < len(arr[0]):
        for i in range(len(arr[0]) - len(arr)):
            answer.append([0] * len(arr[0]))
    
    return answer